src/System.Linq/tests/GroupByTests.cs

```diff
@@ -862,5 +864,28 @@ public static void GroupingKeyIsPublic()
             PropertyInfo key = grouptype.GetProperty("Key", BindingFlags.Instance | BindingFlags.Public);
             Assert.NotNull(key);
         }
+
+        [Theory]
+        [MemberData(nameof(DebuggerAttributesValid_Data))]
+        public void DebuggerAttributesValid<TKey, TElement>(IGrouping<TKey, TElement> grouping, string expected, TKey dummy1, TElement dummy2)
+        {
+            // The dummy parameters can be removed once https://github.com/dotnet/buildtools/pull/1300 is brought in.
+            Assert.Equal(expected, DebuggerAttributes.ValidateDebuggerDisplayReferences(grouping));
```

Comment:
A better name for this function now would be something like `EvaluateDebuggerDisplayReferences`